specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1082 @@
+
+Subnets are identified with a unique ID that is inferred deterministically from the ID of its ancestor and from the ID of the `SA` that governs the subnet's operation. This deterministic naming enables the discovery of --- and interaction with --- subnets from any other point in the hierarchy without the need of a discovery service: peers only need to send a message to the subnet's specific pubsub topic, identified with the subnet's ID.
+
+For a subnet to be registered in the `SCA`, the actor needs to send a new message to the `Register()` function of the `SCA`. This transaction includes the amount of tokens the subnet wants to add as collateral in the parent chain to secure the child chain. The `SA` may implement custom policies that needs to be fulfilled before registering the subnet and triggering `Register()` message to the `SCA`, like for instance waiting for a minimum number of validators to join the network and put some collateral. The `SA` will also need to require that enough collateral is staked before registering to ensure that the message succeeds. For a subnet to be activated in HC, at least `CollateralThreshold` needs to be staked in the `SCA`.
+
+### Leaving and killing a subnet
+Members of a subnet can leave the subnet at any point by sending a message to the subnet's `SA` in the parent chain. If the miner fulfills the requirements to leave the subnet defined in the subnet's `SA` when it was deployed, a message to the `SCA` is triggered by the `SA` to release the miner's collateral.  If a validator leaving the subnet brings the collateral of the subnet below `CollateralThreshold`, the subnet gets in an `Inactive` state, and it can no longer interact with the rest of chains in the hierarchy or checkpoint to the top chain. To recover its `Active` state, any participant of the protocol (within or outside the subnet, user or validator) need to put up additional collateral.
+
+Validators of a subnet can also kill a subnet by sending a `ReleaseCollateral()` message to the `SA`. Similar to the previous situation, the `SA` sends a message to the `SCA` to release all the collateral for the subnet if all the requirements to kill the subnet are fulfilled.
```

Comment:
The `ReleaseCollateral` message appears in the SCA interface, not the SA, and it takes an amount to be released. Is it really meant to kill the subnet? I thought it might deactivate it, but the SA will have to receive the `Kill` message.